### DISCUSSION TITLE
Component won't be displayed in Next.js

### DISCUSSION THREAD

**Discussion by mihaimazareanu** [neuefische/questions#335] [Web] [answered]
Hi everyone,

For my capstone project I am trying to create a button that contains a SVG as text content. When I wrote the code directly in index.js, everything was fine, but when I tried to make a component out of that button, the component won't be displayed. Any ideas?

The component is called RegisterButton: 

    import Head from "next/head";
    import styled from "styled-components";
    import {useState} from "react";
    import RegisterButton from "../../components/register-button/RegisterButton";
    
    export default function HomePage() {
    return (
      <>
        <Head />
        <Heading>kidsFi - Finance for kids</Heading>
          <RegisterButton
            register={register}
            handleClickRegister={handleClickRegister}
          />
      </>

Heading is just a h1 styled component.

And this is the component function:

    import styled from "styled-components";
    
    export default function RegisterButton({handleClickRegister}) {
    const onClickRegister = () => {
    handleClickRegister();
    };
    <RegButton onClick={onClickRegister}>
    <svg
      className="change-color"
      title="signin-icon"
      width="53"
      height="51"
      viewBox="0 0 53 51"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      role="img"
    >
      <path
        d="M33.125 8.83337C30.7823 8.83337 28.5355 9.76403 26.8789 11.4206C25.2224 13.0772 24.2917 15.324 24.2917 17.6667C24.2917 20.0095 25.2224 22.2562 26.8789 23.9128C28.5355 25.5694 30.7823 26.5 33.125 26.5C35.4678 26.5 37.7146 25.5694 39.3711 23.9128C41.0277 22.2562 41.9584 20.0095 41.9584 17.6667C41.9584 15.324 41.0277 13.0772 39.3711 11.4206C37.7146 9.76403 35.4678 8.83337 33.125 8.83337ZM33.125 13.0292C33.734 13.0292 34.3371 13.1492 34.8997 13.3822C35.4624 13.6153 35.9736 13.9569 36.4042 14.3875C36.8349 14.8181 37.1765 15.3294 37.4095 15.892C37.6426 16.4547 37.7625 17.0577 37.7625 17.6667C37.7625 18.2757 37.6426 18.8788 37.4095 19.4414C37.1765 20.004 36.8349 20.5153 36.4042 20.9459C35.9736 21.3765 35.4624 21.7181 34.8997 21.9512C34.3371 22.1843 33.734 22.3042 33.125 22.3042C32.516 22.3042 31.913 22.1843 31.3503 21.9512C30.7877 21.7181 30.2765 21.3765 29.8458 20.9459C29.4152 20.5153 29.0736 20.004 28.8405 19.4414C28.6075 18.8788 28.4875 18.2757 28.4875 17.6667C28.4875 16.4368 28.9761 15.2572 29.8458 14.3875C30.7155 13.5178 31.8951 13.0292 33.125 13.0292ZM8.83337 15.4584V22.0834H2.20837V26.5H8.83337V33.125H13.25V26.5H19.875V22.0834H13.25V15.4584H8.83337ZM33.125 28.7084C27.2288 28.7084 15.4584 31.6455 15.4584 37.5417V44.1667H50.7917V37.5417C50.7917 31.6455 39.0213 28.7084 33.125 28.7084ZM33.125 32.9042C39.6838 32.9042 46.5959 36.1284 46.5959 37.5417V39.9709H19.6542V37.5417C19.6542 36.1284 26.5 32.9042 33.125 32.9042Z"
        fill="#5E8C49"
      />
    </svg>
    </RegButton>;
    }
    const RegButton = styled.button`
    border: none;
    `

I would be thankful for your input on this issue.

**Comment by actyralf** [comment, marked as answer]
I think you forgot the `return` statement in your `RegisterButton` component.

**Comment by mihaimazareanu** [comment]
Yup, that was it. 🙈 No wonder I couldn't find any similar discussions,
since the solution was so simple. 🤦🏻 Thank you for the late reply, Ralf.
🙏
